File: app/Http/Controllers/SpecialistPatternsJava/Structural/Adapter/AdapterJava/src/RussianWeather.java
public class RussianWeather implements WeatherService {

    String city; // Поле для хранения текущего местоположения

    /**
     * Метод для получения температуры
     * @return Температура в градусах Цельсия
     */
    @Override
    public double getTemperature() {
        switch(city) {
            case "Москва"  : return 25;
            case "Санкт-Петербург" : return 18;
            default : return 20;
        }
    }

    /**
     * Метод для получения скорости ветра
     * @return Скорость ветра в метрах в секунду
     */
    @Override
    public double getWind() {
        switch(city) {
            case "Москва"  : return 5;
            case "Санкт-Петербург" : return 13;
            default : return 1;
        }
    }

    /**
     * Метод для получения ощущаемой температуры
     * @return Ощущаемая температура в градусах Цельсия
     */
    @Override
    public double getFeelsLikeTemperature() {
        switch(city) {
            case "Москва"  : return 23;
            case "Санкт-Петербург" : return 16;
            default : return 20;
        }
    }

    /**
     * Метод для установки текущего местоположения
     * @param city Название города, для которого нужно получить погоду
     */
    @Override
    public void setPosition(String city) {
        this.city = city;
    }
}

File: app/Http/Controllers/SpecialistPatterns/Adapter/AdapterClass/RussianWeather.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\Adapter\AdapterClass;


class RussianWeather implements WeatherService
{
    // Свойство для хранения города
    private string $city;

    /**
     * Метод для получения температуры
     */
    public function getTemperature(): float
    {
        if ($this->city == "Москва") return 25.0;
        if ($this->city == "Санкт-Петербург") return 18.0;
        return 20.0;
    }

    /**
     * Метод для получения скорости ветра в м/с
     */
    public function getWind(): float
    {
        if ($this->city == "Москва") return 5.0;
        if ($this->city == "Санкт-Петербург") return 13;
        return 1;
    }

    /**
     * Метод для получения ощущаемой температуры
     */
    public function getFeelsLikeTemperature(): float
    {
        if ($this->city == "Москва") return 23;
        if ($this->city == "Санкт-Петербург") return 16;
        return 20;
    }

    /**
     * Виртуальный метод для установки местоположения (города)
     *
     * @param string $city
     */
    public function setPosition(string $city): void
    {
        $this->city = $city;
    }
}
